components/com_content/tmpl/form/edit.php

```diff
@@ -44,7 +47,7 @@
     </div>
     <?php endif; ?>
 
-    <form action="<?php echo Route::_('index.php?option=com_content&a_id=' . (int) $this->item->id); ?>" method="post" name="adminForm" id="adminForm" class="form-validate form-vertical">
+    <form action="<?php echo Route::_('index.php?option=com_content&a_id=' . (int) $this->item->id . ($menu ? "&Itemid={$menu->id}" : '')); ?>" method="post" name="adminForm" id="adminForm" class="form-validate form-vertical">
```

Comment:
```suggestion
    <form action="<?php echo Route::_('index.php')); ?>" method="post" name="adminForm" id="adminForm" class="form-validate form-vertical">
```
Unfortunately you can't add the Itemid in this way and shouldn't really have to do that ever really. See above for a better solution. This will result in the current request being the form action.